crates/iota-node/src/lib.rs

```diff
@@ -2056,9 +2057,20 @@ pub async fn build_http_server(
             ))?;
         }
 
-        // TODO: Init from chain if config is not set once `IotaNamesConfig::from_chain`
-        // is implemented
-        let iota_names_config = config.iota_names_config.clone().unwrap_or_default();
+        let iota_names_config = match &config.iota_names_config {
+            Some(config) => config.clone(),
+            None => {
+                if let Ok(config) = IotaNamesConfig::from_env() {
+                    config
+                } else {
+                    let chain = state
+                        .get_chain_identifier()
+                        .ok_or(IotaError::Storage("chain identifier not found".to_string()))?
+                        .chain();
+                    IotaNamesConfig::from_chain(&chain)
```

Comment:
I don't think this is safe to do right now.
If there is no config provided or env variables this will fail because of the `todos`? not. Until we have fully implemented `from_chain` we should provide the default config.

Reply:
@DaughterOfMars just comment https://github.com/iotaledger/iota/blob/8c9c961e589ffba62a743acbbd2035771e860056/crates/iota-names/src/config.rs#L57-L58 then it should be fine I guess